### ISSUE TITLE
tainted

### ISSUE THREAD

**Issue by L3n1n** [p3lim-wow/BetterWorldQuests#24]
Sometimes i have notice about blocking interface and i can't use action buttons because interface tainted. It happens all the time but i can't describe hot to reproducing the bug. There is my taint.log:

```
12/4 19:11:29.810  Global variable isActivitySet tainted by BetterWorldQuests - Interface/FrameXML/WorldMapBountyBoard.lua:152 Frame:Refresh()
12/4 19:11:29.810      Interface/AddOns/Blizzard_WorldMap/Blizzard_WorldMap.lua:285 WorldMapFrame:RefreshOverlayFrames()
12/4 19:11:29.810      Interface/AddOns/Blizzard_WorldMap/Blizzard_WorldMap.lua:233
12/4 19:11:29.810      WorldMapFrame:OnMapChanged()
12/4 19:11:29.810      Interface/AddOns/Blizzard_MapCanvas/Blizzard_MapCanvas.lua:60 WorldMapFrame:SetMapID()
12/4 19:11:29.810      Interface/AddOns/Blizzard_MapCanvas/Blizzard_MapCanvas.lua:789 WorldMapFrame:NavigateToParentMap()
12/4 19:11:29.810      Interface/AddOns/Blizzard_MapCanvas/MapCanvas_ScrollContainerMixin.lua:127
12/4 19:11:29.810  Execution tainted by BetterWorldQuests while reading isActivitySet - Interface/FrameXML/WorldMapBountyBoard.lua:591 Button:Refresh()
12/4 19:11:29.810      Interface/AddOns/Blizzard_WorldMap/Blizzard_WorldMap.lua:285 WorldMapFrame:RefreshOverlayFrames()
12/4 19:11:29.810      Interface/AddOns/Blizzard_WorldMap/Blizzard_WorldMap.lua:233
12/4 19:11:29.810      WorldMapFrame:OnMapChanged()
12/4 19:11:29.810      Interface/AddOns/Blizzard_MapCanvas/Blizzard_MapCanvas.lua:60 WorldMapFrame:SetMapID()
12/4 19:11:29.810      Interface/AddOns/Blizzard_WorldMap/Blizzard_WorldMap.lua:243
12/4 19:11:29.810      WorldMapFrame:Show()
12/4 19:11:29.810      Interface/FrameXML/UIParent.lua:2853 Frame:SetUIPanel()
12/4 19:11:29.810      Interface/FrameXML/UIParent.lua:2696 Frame:ShowUIPanel()
12/4 19:11:29.810      Interface/FrameXML/UIParent.lua:2561
12/4 19:11:29.810      Frame:SetAttribute()
12/4 19:11:29.810      Interface/FrameXML/UIParent.lua:3159 ShowUIPanel()
12/4 19:11:29.810      Interface/FrameXML/QuestLogOwnerMixin.lua:103 WorldMapFrame:SetDisplayState()
12/4 19:11:29.810      Interface/FrameXML/QuestLogOwnerMixin.lua:41 WorldMapFrame:HandleUserActionToggleSelf()
12/4 19:11:29.810      Interface/AddOns/Blizzard_WorldMap/Blizzard_WorldMap.lua:440 ToggleWorldMap()
12/4 19:11:29.810      TOGGLEWORLDMAP:1
12/4 19:11:29.810  Interface/FrameXML/ActionButton.lua:1071 UpdateFlyout()
12/4 19:11:29.810  An action was blocked because of taint from BetterWorldQuests - 
12/4 19:11:29.810      UseAction()
12/4 19:11:29.810      Interface/FrameXML/SecureTemplates.lua:364 handler()
12/4 19:11:29.810      Interface/FrameXML/SecureTemplates.lua:690 PerformAction()
12/4 19:11:29.810      Interface/FrameXML/SecureTemplates.lua:704 OnActionButtonClick()
12/4 19:11:29.810      Interface/FrameXML/SecureTemplates.lua:746 SecureActionButton_OnClick()
12/4 19:11:29.810      Interface/FrameXML/ActionButton.lua:109 TryUseActionButton()
12/4 19:11:29.810      Interface/FrameXML/ActionButton.lua:144 ActionButtonDown()
12/4 19:11:29.810      ACTIONBUTTON2:2
```

**Comment by p3lim** [comment]
This is actually a Blizzard issue, tracking it here: https://github.com/Stanzilla/WoWUIBugs/issues/366

I'll test out a temporary fix for it.

**Comment by p3lim** [comment]
Having trouble reproducing this though, can you consistently reproduce this somehow?

**Comment by p3lim** [comment]
Blizzard fixed the global leak in build 10.0.2.46999, will verify when it's live for my region.

**Comment by p3lim** [comment]
Can't reproduce it as of build 47067, which also included a bunch of other taint fixes.
